mongo/mongo.go

```diff
@@ -89,3 +90,25 @@ func (m *Mongo) GetBooks() (models.Books, error) {
 
 	return *books, nil
 }
+
+// Checker calls an api health endpoint and updates the provided CheckState
+func (m *Mongo) Checker(ctx context.Context, state *dpHealthcheck.CheckState) error {
+	if err := m.Healthcheck(ctx); err != nil {
+		state.Update(dpHealthcheck.StatusCritical, err.Error(), 0)
+		return nil
```

Comment:
Don't you want be returning `err` in this instance?

Reply:
😱 yes!

Reply:
Actually there is no need to return the error. The context here is to capture the fact that the application cannot communicate to mongo (via the client c) and therefore health state is updated, we can now ignore the error.

Reply:
Also this is a duplication of the code in dp-mongodb, you could re-use the library? That way you will inherit the collection check too.

Reply:
I've made changes to use the library now 🙂 
Thanks for the feedback!